src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPGlobalProperties.cs

```diff
@@ -0,0 +1,392 @@
+
+            string udp6FileContents = File.ReadAllText(NetworkFiles.Udp6ConnectionsFile);
+            string[] v6connections = udp6FileContents.Split(s_newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            IPEndPoint[] endPoints = new IPEndPoint[v4connections.Length + v6connections.Length - 2]; // First line is header in each file
+            int index = 0;
+
+            // UDP Connections
+            for (int i = 1; i < v4connections.Length; i++) // Skip first line header
+            {
+                string line = v4connections[i];
+                IPAddress remoteIPAddress;
+                int remotePort;
+                ParseRemoteConnectionInformation(line, out remoteIPAddress, out remotePort);
+
+                endPoints[index++] = new IPEndPoint(remoteIPAddress, remotePort);
+            }
+
+            // UDP6 Connections
+            for (int i = 1; i < v6connections.Length; i++) // Skip first line header
+            {
+                string line = v6connections[i];
+                IPAddress remoteIPAddress;
+                int remotePort;
+                ParseRemoteConnectionInformation(line, out remoteIPAddress, out remotePort);
+
+                endPoints[index++] = new IPEndPoint(remoteIPAddress, remotePort);
+            }
+
+            return endPoints;
+        }
+
+        public override IcmpV4Statistics GetIcmpV4Statistics()
+        {
+            return LinuxIcmpV4Statistics.CreateIcmpV4Statistics();
+        }
+
+        public override IcmpV6Statistics GetIcmpV6Statistics()
+        {
+            return LinuxIcmpV6Statistics.CreateIcmpV6Statistics();
+        }
+
+        public override IPGlobalStatistics GetIPv4GlobalStatistics()
+        {
+            return LinuxIPGlobalStatistics.CreateIPv4Statistics();
+        }
+
+        public override IPGlobalStatistics GetIPv6GlobalStatistics()
+        {
+            return LinuxIPGlobalStatistics.CreateIPv6Statistics();
+        }
+
+        public override TcpStatistics GetTcpIPv4Statistics()
+        {
+            return new LinuxTcpStatistics(ipv6:false);
+        }
+
+        public override TcpStatistics GetTcpIPv6Statistics()
+        {
+            return new LinuxTcpStatistics(ipv6:true);
+        }
+
+        public override UdpStatistics GetUdpIPv4Statistics()
+        {
+            return LinuxUdpStatistics.CreateUdpIPv4Statistics();
+        }
+
+        public override UdpStatistics GetUdpIPv6Statistics()
+        {
+            return LinuxUdpStatistics.CreateUdpIPv6Statistics();
+        }
+
+        public override UnicastIPAddressInformationCollection GetUnicastAddresses()
+        {
+            UnicastIPAddressInformationCollection collection = new UnicastIPAddressInformationCollection();
+            foreach (UnicastIPAddressInformation info in
+                LinuxNetworkInterface.GetLinuxNetworkInterfaces().SelectMany(lni => lni.GetIPProperties().UnicastAddresses))
```

Comment:
Design: this introduced a new dependency on Linq which S.N.NetworkInformation doesn't have on Windows.
@ericstj @stephentoub, @pgavlin: Are there any specific guidelines against using new dependencies? 
I'm personally not against using Linq but this could have some interesting package dependency implications that could artificially grow the application size (maybe even on Windows?)